### PR TITLE
Test for the Rln relay/membership management contract deployment

### PR DESCRIPTION
**Pull request by staheri14**
This is going to be a partial progress for https://github.com/status-im/nim-waku/issues/296
Which implements a test for the deployment of the membership contract and interacting with its interface 
- [x] Get the contract compiled and deployed
- [X] Implement the contract interface
- [X] Implement and test the interaction with the contract interface

### CODE REVIEW

**Comment by oskarth** [comment]
@staheri14 would you mind rebasing this on master? Makes it easier to see diff. It also means less merge conflicts to resolve later!

**Comment by oskarth** [comment]
Also wdyt of a smaller PR to introduce just basic web3 dependency? I need this as well so if we could merge this to master it might be useful.

**Comment by staheri14** [comment]
> Also wdyt of a smaller PR to introduce just basic web3 dependency? I need this as well so if we could merge this to master it might be useful.

Sure! makes sense, opened the PR https://github.com/status-im/nim-waku/pull/316

**Comment by staheri14** [comment]
> @staheri14 would you mind rebasing this on master? Makes it easier to see diff. It also means less merge conflicts to resolve later!

Thanks! done!

**Comment by oskarth** [comment]
Would it be possible to get this merged as partial progress? I'm doing something very similar as well.

**Comment by staheri14** [comment]
> Would it be possible to get this merged as partial progress? I'm doing something very similar as well.

Sure, ready for merge!

File: tests/v2/test_waku_store.nim [hunk 1]
```diff
@@ -36,12 +36,12 @@ procSuite "Waku Store":
 
     proto.setPeer(listenSwitch.peerInfo)
 
-    var subscriptions = newTable[string, MessageNotificationSubscription]()
-    subscriptions["test"] = subscription
+    var subscriptions = newTable[string, MessageNotificationSubscription]() # listener
```
Comment (oskarth) [inline tests/v2/test_waku_store.nim:39]: All the changes in this file seems unnecessary

**Review by oskarth — APPROVED** [review]
LGTM but please remove misc notes store changes

**Comment by oskarth** [comment]
@staheri14 what's the blocker for enabling this test by default?

File: tests/v2/test_waku_store.nim (hunk 1 above)
Comment (staheri14) [reply tests/v2/test_waku_store.nim]: Sure, addressed

**Comment by staheri14** [comment]
> @staheri14 what's the blocker for enabling this test by default?

Not exactly a blocker, but it requires an Eth client (emulator) like ganache to be installed and run on  localhost:8545, otherwise the test fails. At the moment I was not sure how to enforce this (maybe through docker or .nims/Makefile). Will figure this out.

**Comment by oskarth** [comment]
@staheri14 please rebase and squash for merging :)

**Pull request merged**